fundingmanager.go

```diff
@@ -1283,8 +1335,14 @@ func (f *fundingManager) waitForFundingWithTimeout(completeChan *channeldb.OpenC
 			// The fundingManager is shutting down, will resume
 			// waiting for the funding transaction on startup.
 			return
-		case <-waitingDoneChan:
-			close(doneChan)
+		case shortChanID, ok := <-waitingConfChan:
+			if !ok {
+				// Failed waiting for confirmation, close
+				// confChan to indicate failure.
+				close(confChan)
+				return
+			}
+			confChan <- shortChanID
```

Comment:
In order to avoid possible future deadlocks on shutdown, this should be converted to instead be a conditional send:
```go
select {
   case confChan <- shortChanID:
   case <-f.quit:
        return
}
```